### ISSUE TITLE
need findOne callback

### ISSUE THREAD

**Issue by bevinhex** [meteor/meteor#6157]
Here is my problem, I installed summernote:standalone package, then I used react, I wanted to make a component using jsx, I can save summernote data on blur event, but when I try to use 
```
data = statementCollection.findOne();
console.log(data);
$('#summernote').summernote('code',data.text);
```
data is always **null**, I tried onReady callback from subscribe, it seems, it is ready even when data is not ready(null);
I even tried Tracer.autorun to know when my data is valid, no luck.
Googled for about a day, no way of knowing when my data gets ready.
It's driving me nuts!
If only I could have some kind of callback when data ready, my problems would be really easy to solve

**Comment by crapthings** [comment]
try this one

https://github.com/matb33/meteor-collection-hooks

**Comment by bevinhex** [comment]
I've tried that already, the hook is something get triggered whenever the data is modified, meaning, whenever my data changed from somewhere else in the code, the hook triggers;
I don't want that, I just need to know , when the findOne gets my data (success or failure). I only want to call the summernote function one time - the first time I find the record.

**Comment by crapthings** [comment]
why not just use Template.rendered ?

template is just rendered once after u get data

are you using blaze ?

something like

    Template.summernote.rendered = ->
        data = statementCollection.findOne()
        ($ '#summernote').summernote 'code',data.text if data.text?

**Comment by bevinhex** [comment]
I can call summernote function like that?

**Comment by crapthings** [comment]
yes its good to put jquery stuff with template rendered

**Comment by bevinhex** [comment]
Let me check rendered callback first

**Comment by bevinhex** [comment]
I don't think it is going to work, because ```if(data.text)``` will fail, and there is no guarentee that the code will rerun once the data changes

**Comment by bevinhex** [comment]
I just don't understand, why data still not ready inside subcribe callback?

**Comment by crapthings** [comment]
you have to tell its ready when theres no data at all

**Comment by bevinhex** [comment]
I returned [] when no data. I also tried to put breakpoint in subscribe callback, then I can see data is not undefined, but I'll get undefined if I remove the breakpoint. 
It seems, data was ready before I can open the debug window and watch it, but not ready if it runs itself, guess not enough time to be ready

**Comment by crapthings** [comment]
i've wrote an example for u @bevinhex check this

https://github.com/crapthings/mi6157

**Comment by bevinhex** [comment]
Thanks, I got the idea, you mean I can put the js code in render method, so that every time the data gots update, the interface gets update, just like the FLUX idea
What got me in problem was that, I was trying to update the data from the interface, whenever I change the interface data, it updates the data.
I guest 
Interface(trigger event)-> (update)data -> (update)interface
idea works well for solving my problem

To sum up, put ui updating code inside render method if not want extra trouble :-)

**Comment by bevinhex** [comment]
That solved my problem, but I still wonder why the subscribe ready even is fired even when the data is not.

**Comment by crapthings** [comment]
maybe publish this.stop()?

**Comment by bevinhex** [comment]
Not sure, I'll do some experiment about it, then will post the result.

**Comment by tmeasday** [comment]
Hey all, I think this kind of issue using Meteor is best resolved via Stack Overflow or on the forums where others can find it easily.
